### DISCUSSION TITLE
Is it possible to use two different "projects.json" files, one for work projects and one for personal?

### DISCUSSION THREAD

**Discussion by jdkolassa** [alefragnani/vscode-project-manager#609] [Q&A] [answered]
So this might not be a Project Manager specific question and a more general VS Code one, but since my use case is about Project Manager I'll ask here first. 

## My objective
To keep all of my extensions across work and personal computers, _except_ to have a different projects list in my Project Manager extension for work projects and for personal projects. 

### The problem as I see it

I have two personal computers (a desktop and a laptop.) I also have a work computer. I want to use settings sync to sync *all* settings between my personal computers and all but _one_ setting between my personal and work computers -- the `projects.json` file for Project Manager. I do not want all my personal projects to appear in Project Manager while I am working, because there are many learning and weird experiments that I just don't want cluttering up my view. I also don't want to accidentally open up a personal project on my work computer, or a work project on my home computer -- I just want to keep these realms as separate as possible. 

I'm not sure if there is a way to set up VS Code to use two slightly different profiles -- I have seen the idea of [code profiles](https://dev.to/jsjoeio/how-to-create-code-profiles-in-vscode-3ofo) and I have installed Profile Switcher -- but I don't understand if this can accomplish my objective. I thought if I set the following: 

    `{ "projectManager.projectsLocation": "~\\.vscode\\projects.json"}`

Then I could keep separate `projects.json` files on my computers...but then wouldn't that get overridden by a settings sync anyways? 

As for the question "Why not use tags?" the thing is that I plan on using tags for different languages and frameworks, as well as training and testing projects. It's already probably going to be cluttered as-is, so I want a level of organization _above_ that. Is what I am asking for feasible, and if so, do you have any recommendations/suggestions about how to go about it? (Is it just as simple as taking the above setting and changing it inside a code profile? Or is it much more than that?)

**Comment by alefragnani** [comment, marked as answer]
Hi @jdkolassa ,

To summarize, _this is not possible_, not only because the Project Manager extension but also VS Code itself. 

I see three different approaches and their dependencies.

## User Profiles

The extension has no _profile_ concept, neither a way to split projects/configs between personal and work projects. Both, the  `projects.json` file (where the Favorite projects are stored) and the `projectManager.xxx.baseFolders` settings works for a single user/profile.

This _could change_ depending on how the new [User Profiles](https://github.com/microsoft/vscode/issues/116740) feature (which the VS Code team has just started) would work. If the User Profile supports not only user settings but also how extensions manages state, maybe that could work _out of the box_.

But you will have to wait until the feature is fully developed by the VS Code team.

## Tags

If you only care about _Favorite Projects_, I would say `Tags` would work perfectly, because you would need add a single `Tag` to split between Personal and Work projects. `Tags` are not supported on auto-detected projects, yet, but it's on the roadmap.

## Settings Sync

If you rely on this VS Code built in feature, you are out of luck as well. The VS Code team still didn't provided an API for extensions to sync additional files (`projects.json` on this case), so you your projects won't be synced between different computers


In the end, right now, the best approach may be `Tags`, but its not perfect, as you already mentioned.

Hope this helps

**Reply by jdkolassa** [reply]
Hi @alefragnani,

If the extension doesn't sync the `projects.json` file, and that is only for organizing your projects on one computer, I am _perfectly_ fine with that. No complaints. 

Looks like this will work for me! Thanks for the extension, it's great and it's definitely going to come in handy at my office!
